src/addrman.cpp

```diff
@@ -699,3 +700,100 @@ std::vector<bool> CAddrMan::DecodeAsmap(fs::path path)
     }
     return bits;
 }
+
+void CAddrMan::ResetI2PPorts()
+{
+    for (int bucket = 0; bucket < ADDRMAN_NEW_BUCKET_COUNT; ++bucket) {
+        for (int i = 0; i < ADDRMAN_BUCKET_SIZE; ++i) {
+            const auto id = vvNew[bucket][i];
```

Comment:
f108193c6381b4d It looks like the loops can be accelerated with a faster iterator comparison and by avoiding integer conversions. 

<details><summary>suggestion</summary><p>

```diff
diff --git a/src/addrman.cpp b/src/addrman.cpp
index f2d54c0adb..41375e8320 100644
--- a/src/addrman.cpp
+++ b/src/addrman.cpp
@@ -703,8 +703,8 @@ std::vector<bool> CAddrMan::DecodeAsmap(fs::path path)
 
 void CAddrMan::ResetI2PPorts()
 {
-    for (int bucket = 0; bucket < ADDRMAN_NEW_BUCKET_COUNT; ++bucket) {
-        for (int i = 0; i < ADDRMAN_BUCKET_SIZE; ++i) {
+    for (size_t bucket = 0; bucket != ADDRMAN_NEW_BUCKET_COUNT; ++bucket) {
+        for (size_t i = 0; i != ADDRMAN_BUCKET_SIZE; ++i) {
             const auto id = vvNew[bucket][i];
             if (id == -1) {
                 continue;
@@ -728,7 +728,7 @@ void CAddrMan::ResetI2PPorts()
             // re-evaluate that decision, but even if we could, CAddrInfo::GetNewBucket() does not
             // use CAddrInfo::GetKey() so it would end up in the same bucket as before the port
             // change.
-            const auto i_target = addr_info_newport.GetBucketPosition(nKey, true, bucket);
+            const size_t i_target = addr_info_newport.GetBucketPosition(nKey, true, bucket);
 
             if (i_target == i) { // No need to re-position.
                 addr_info = addr_info_newport;
@@ -743,8 +743,8 @@ void CAddrMan::ResetI2PPorts()
         }
     }
 
-    for (int bucket = 0; bucket < ADDRMAN_TRIED_BUCKET_COUNT; ++bucket) {
-        for (int i = 0; i < ADDRMAN_BUCKET_SIZE; ++i) {
+    for (size_t bucket = 0; bucket != ADDRMAN_TRIED_BUCKET_COUNT; ++bucket) {
+        for (size_t i = 0; i != ADDRMAN_BUCKET_SIZE; ++i) {
             const auto id = vvTried[bucket][i];
             if (id == -1) {
                 continue;
@@ -763,8 +763,8 @@ void CAddrMan::ResetI2PPorts()
             // bucket and a position within that bucket. So a re-bucketing may be necessary.
             addr_info_newport.port = I2P_SAM31_PORT;
 
-            const auto bucket_target = addr_info_newport.GetTriedBucket(nKey, m_asmap);
-            const auto i_target = addr_info_newport.GetBucketPosition(nKey, false, bucket_target);
+            const size_t bucket_target = addr_info_newport.GetTriedBucket(nKey, m_asmap);
+            const size_t i_target = addr_info_newport.GetBucketPosition(nKey, false, bucket_target);
 
             if (bucket_target == bucket && i_target == i) { // No need to re-position.
                 addr_info = addr_info_newport;
```
</p></details>

Reply:
(This isn't a hotspot but is something that could also be considered in places that are.)

Reply:
Will leave it up to @vasild but it seems to me that integer and iterator conversions seem an unlikely source of performance problems.

> (This isn't a hotspot but is something that could also be considered in places that are.)

Sure, it might have impact on tight loops that loop over large memory areas.

Reply:
1.
`ADDRMAN_NEW_BUCKET_COUNT` is of type `int`.

```cpp
for (int bucket = 0; bucket < ADDRMAN_NEW_BUCKET_COUNT; ++bucket) {
```

how is changing `bucket` to `size_t` going to make it faster? (otoh, it could produce some warning about comparing `size_t` to `int`)

2.
Current code:
```cpp
int GetFoo();
auto x = GetFoo();
```
So `x` will be defined as type `int`. How will explicitly defining it as `size_t` make it faster? (otoh, assigning from `int` to `size_t` could cause some compiler to warn about it)

I tried this:

```cpp
#define N (1 << 10)

void foo()
{
    int array[N];
    for (int i = 0; i < N; ++i) {
        printf("%d\n", array[i]);
    }
}
```

Changing `i` from `int` to `size_t` produces **identical** assembler output:

```asm
(gdb) disas foo
Dump of assembler code for function _Z3foov:
   0x0000000000202220 <+0>:	push   %rbp
   0x0000000000202221 <+1>:	mov    %rsp,%rbp
   0x0000000000202224 <+4>:	push   %rbx
   0x0000000000202225 <+5>:	sub    $0x1008,%rsp
   0x000000000020222c <+12>:	xor    %ebx,%ebx
   0x000000000020222e <+14>:	xchg   %ax,%ax
   0x0000000000202230 <+16>:	mov    -0x1010(%rbp,%rbx,4),%esi
   0x0000000000202237 <+23>:	mov    $0x200cc7,%edi
   0x000000000020223c <+28>:	xor    %eax,%eax
   0x000000000020223e <+30>:	call   0x202670 <printf@plt>
   0x0000000000202243 <+35>:	add    $0x1,%rbx
   0x0000000000202247 <+39>:	cmp    $0x400,%rbx
   0x000000000020224e <+46>:	jne    0x202230 <_Z3foov+16>
   0x0000000000202250 <+48>:	add    $0x1008,%rsp
   0x0000000000202257 <+55>:	pop    %rbx
   0x0000000000202258 <+56>:	pop    %rbp
   0x0000000000202259 <+57>:	ret    
End of assembler dump.
```